core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncSigningStage.java

```diff
@@ -232,10 +233,23 @@ private boolean shouldSign(Signer signer) {
     }
 
     /**
-     * Returns true if we should use the selected out scheme attribute for signing.
+     * Returns true if we should use SRA signing logic.
      */
-    private boolean shouldUseSelectedAuthScheme(RequestExecutionContext context) {
-        return context.executionAttributes().getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME) != null;
+    private boolean shouldDoSraSigning(RequestExecutionContext context) {
+        return !isSignerOverridden(context)
+               && context.executionAttributes().getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME) != null;
+    }
+
+    // This is copied from SignerOverrideUtils.isSignerOverridden, but that is in aws-core, and this is in sdk-core.
+    // Just this method could be moved to sdk-core somewhere, but the other methods in SignerOverrideUtils depend on AwsRequest
+    // and AwsRequestOverrideConfiguration which are in aws-core. Just duplicating the logic here seemed ok.
```

Comment:
I'm considering moving just this method to a `SignerOverrideUtils` in sdk-core. I would have renamed the one in aws-core to AwsSignerOverrideUtils maybe, but I can't change the name / remove the current method from SdkProtectedApi, so I'll probably duplicate just this method and add a `SdkSignerOverrideUtils` in sdk-core. SignerOverrideUtils in aws-core actually should get deprecated in subsequent PR.

Reply:
I found an issue with the existing method in SignerOverrideUtils which just checks that the ExecutionAttribute SIGNER_OVERRIDDEN is present and is not checking that it is `true`. It doesn't affect anything since the attribute is never set to false anywhere. So I'm going to pull this out in a new Utils in sdk-core.